src/algebraic_geometry/prime_spectrum/basic.lean

```diff
@@ -51,25 +54,73 @@ universes u v
 
 variables (R : Type u) [comm_ring R]
 
-/-- The prime spectrum of a commutative ring `R`
-is the type of all prime ideals of `R`.
+/-- The maximal spectrum of a commutative ring `R` is the type of all prime ideals of `R`. -/
+@[nolint has_nonempty_instance] def maximal_spectrum := {I : ideal R // I.is_maximal}
+
+namespace maximal_spectrum
+
+variable {R}
+
+/-- A method to view a point in the maximal spectrum of a commutative ring
+as an ideal of that ring. -/
+abbreviation as_ideal (x : maximal_spectrum R) : ideal R := x.val
+
+instance is_maximal (x : maximal_spectrum R) : x.as_ideal.is_maximal := x.property
+
+instance is_prime (x : maximal_spectrum R) : x.as_ideal.is_prime := x.is_maximal.is_prime
+
+variables [no_zero_divisors R] {K : Type v} [field K] [algebra R K] [is_fraction_ring R K]
```

Comment:
I think it is better to use `is_domain` here.
We will lose nothing since the trivial ring does not have a fraction ring.

Reply:
```
import ring_theory.localization.fraction_ring
import data.zmod.defs

instance : unique (zmod 1) := fin.unique

example : is_fraction_ring (zmod 1) (zmod 1) :=
have nzd : non_zero_divisors (zmod 1) := ⟨0, λ y h, subsingleton.elim _ _⟩,
{ map_units := λ x, (is_unit_of_dvd_one _ $ ⟨0, subsingleton.elim _ _⟩),
  surj := λ x, ⟨⟨0, nzd⟩, subsingleton.elim _ _⟩,
  eq_iff_exists := λ x y, ⟨λ h, ⟨nzd, subsingleton.elim _ _⟩, λ h, subsingleton.elim _ _⟩ }
```
🙃

Reply:
Shall I revert this?

Reply:
I guess no if you don't need it. The trivial ring isn't field so this won't be applicable as long as `[field K]` is there.